tokio/src/io/interest.rs

```diff
@@ -11,43 +11,59 @@ use std::ops;
 /// I/O resource readiness states.
 #[cfg_attr(docsrs, doc(cfg(feature = "net")))]
 #[derive(Clone, Copy, Eq, PartialEq)]
-pub struct Interest(mio::Interest);
+pub struct Interest {
```

Comment:
Could you change this to a `usize` and define the bits yourself, similar to how `Ready` does it: https://github.com/tokio-rs/tokio/blob/master/tokio/src/io/ready.rs.

I would prefer not growing the struct size and avoid branches in the `is_*` checks.

Reply:
is there a particular reason to choose for `usize`? Maybe because the interest value doesn't live long and is likely in a register, `usize` is no worse than `u8` (which would currently fit, with 2 bits to spare)?

Reply:
`usize` doesn't matter really, it is just what `mio::Interest` uses.

Reply:
mio uses a `NonZeroU8` https://github.com/tokio-rs/mio/blob/master/src/interest.rs#L17

```rust
#[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord)]
pub struct Interest(NonZeroU8);
```